src/dosext/mfs/mfs.c

```diff
@@ -2691,28 +2691,46 @@ static int lock_file_region(int fd, int cmd, struct flock *fl, long long start,
   if (start == 0x100000000LL && config.full_file_locks) {
     start = len = 0;
   }
+
 #ifdef F_GETLK64
-  if (cmd == F_SETLK64 || cmd == F_GETLK64) {
+  if (sizeof(struct flock) != sizeof(struct flock64)) { // specific 64bit
```

Comment:
https://github.com/lattera/glibc/blob/master/bits/fcntl.h#L122
`/* Note this matches struct flock exactly.  */`
So... lets not complicate things, how about just a
`static_assert()` here, then upload to LP and see?
If nothing triggers, we are good.